R/mobr.R

```diff
@@ -244,26 +244,25 @@ rarefaction = function(x, method, effort=NULL) {
 # of individuals) defined by ref_dens, evaluated at specified points (given by
 # inds) Output: a two-column data frame, with sample size
 # (effort) and deltaS (effect of N)
-effect_of_N = function(comm_group, ref_dens, inds){
+deltaS_N = function(comm_group, ref_dens, inds){
   nplots = nrow(comm_group)
   group_dens = sum(comm_group) / nplots
   # calcualte the number of individuals sampled
   # as each of the nplots are collected
   samp_effort = round((1:nplots) * group_dens)
-  #effort = effort[which(effort <= min(sum(comm_group), ref_dens * nrow(comm_group)))]
   # use individual based rarefaction evaluated at the 
   # number of individuals sampled for each plot
   S_samp = rarefaction(comm_group, 'indiv', samp_effort)
   # rescale and interpolate this plot based S to individual based 
   # using the ref_density (i.e., not the observed density)
   rescaled_effort = round(1:nplots * ref_dens)
-  if (max(rescaled_effort) > max(inds)) 
-    warning('Extrapolating the rarefaction curve because the number of rescaled individuals is larger than the inds argument')
-  interp_S_samp = pchip(c(1, rescaled_effort),
-                        c(1, S_samp), inds)
+  if (max(rescaled_effort) < max(inds))
+    warning('Extrapolating the rarefaction curve because the number of rescaled
```

Comment:
Got it, thanks for catching that. I'll get them fixed.

On Sep 4, 2016 7:57 PM, "Dan McGlinn" <notifications@github.com> wrote:

> In R/mobr.R
> <https://github.com/MoBiodiv/mobr/pull/75#discussion_r77460999>:
>
> >    # use individual based rarefaction evaluated at the
> >    # number of individuals sampled for each plot
> >    S_samp = rarefaction(comm_group, 'indiv', samp_effort)
> >    # rescale and interpolate this plot based S to individual based
> >    # using the ref_density (i.e., not the observed density)
> >    rescaled_effort = round(1:nplots * ref_dens)
> > -  if (max(rescaled_effort) > max(inds))
> > -    warning('Extrapolating the rarefaction curve because the number of rescaled individuals is larger than the inds argument')
> > -  interp_S_samp = pchip(c(1, rescaled_effort),
> > -                        c(1, S_samp), inds)
> > +  if (max(rescaled_effort) < max(inds))
> > +    warning('Extrapolating the rarefaction curve because the number of rescaled
>
> Your change here looks good thanks for fixing this. Unfortunately though
> warning is not a very smart function and the carriage return you put
> after rescaled will not be stripped when the string is printed so you have
> to leave the string as one long line of text.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/MoBiodiv/mobr/pull/75/files/58cad734c240d466115a6c2da38ef92405843f7a#r77460999>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ABGoPs71QFXffVvJwtUv6zjQGlvuUjERks5qm1rcgaJpZM4J0cOf>
> .
>